### PR TITLE
add offset and make embedded widget paint itself

### PR DESCRIPTION
**Pull request by renancaraujo**
Before:
ezgif-1-fe98c5795dad

After:
ezgif-1-cdeab3a267d4

REF: #56

### CODE REVIEW

File: lib/game.dart [hunk 1]
```diff
@@ -137,7 +136,7 @@ class _GameRenderBox extends RenderBox with WidgetsBindingObserver {
   @override
   void paint(PaintingContext context, Offset offset) {
     context.canvas.save();
-    context.canvas.translate(game.builder.offset.dx, game.builder.offset.dy);
+    context.canvas.translate(game.builder.offset.dx + offset.dx, game.builder.offset.dy+ offset.dy);
```
Comment (renancaraujo) [inline lib/game.dart]: This line is key. Please, check if this wouldn't break anything @luanpotter

File: lib/game.dart [hunk 2]
```diff
@@ -314,50 +313,45 @@ class SimpleGame extends BaseGame {
 /// Provided it with a [Game] instance for your game and the optional size of the widget.
 /// Creating this without a fixed size might mess up how other components are rendered with relation to this one in the tree.
 /// You can bind Gesture Recognizers immediately around this to add controls to your widgets, with easy coordinate conversions.
-class EmbeddedGameWidget extends StatefulWidget {
+class EmbeddedGameWidget extends LeafRenderObjectWidget{
```
Comment (luanpotter) [inline lib/game.dart]: {

File: lib/game.dart [hunk 3]
```diff
@@ -314,50 +313,45 @@ class SimpleGame extends BaseGame {
...
-    WidgetsBinding.instance.addPostFrameCallback(_afterLayout);
-  }
-
-  @override
-  void didUpdateWidget(EmbeddedGameWidget oldWidget) {
-    super.didUpdateWidget(oldWidget);
-    WidgetsBinding.instance.addPostFrameCallback(_afterLayout);
-  }
+  RenderBox createRenderObject(BuildContext context) {
+    if(size != null){
```
Comment (luanpotter) [inline lib/game.dart]: if (size != null) {

File: lib/game.dart [hunk 4]
```diff
@@ -314,50 +313,45 @@ class SimpleGame extends BaseGame {
...
+      return RenderEmbeddedGameBox(
+          _GameRenderBox(context, game),
+          widgetSize: this.size
```
Comment (luanpotter) [inline lib/game.dart]: wisgetSize: this.size,

File: lib/game.dart [hunk 5]
```diff
@@ -314,50 +313,45 @@ class SimpleGame extends BaseGame {
...
-    RenderBox box = context.findRenderObject();
-    widget.game.builder.offset = box.localToGlobal(Offset.zero);
   }
 
   @override
-  Widget build(BuildContext context) {
-    if (widget.size == null) {
-      return widget.game.widget;
+  void updateRenderObject(BuildContext context, RenderBox renderObject) {
+    if(size != null){
```
Comment (luanpotter) [inline lib/game.dart]: if (size != null) {

File: lib/game.dart [hunk 6]
```diff
@@ -314,50 +313,45 @@ class SimpleGame extends BaseGame {
...
+      RenderEmbeddedGameBox _renderObject = renderObject as RenderEmbeddedGameBox;
+      _renderObject..game = _GameRenderBox(context, game)..widgetSize = this.size;
+    } else {
+      _GameRenderBox _renderObject = renderObject as _GameRenderBox;
+      _renderObject..game = game;
```
Comment (luanpotter) [inline lib/game.dart]: here you can use only a single dot, right?

File: lib/game.dart (hunk 1 above)
Comment (luanpotter) [reply lib/game.dart]: Looks good to me, but please try it out with the examples in the docs folder and also flame-example or bgug!

File: lib/game.dart (hunk 4 above)
Comment (renancaraujo) [reply lib/game.dart]: I don't think wisgetSize wont break anything

File: lib/game.dart (hunk 5 above)
Comment (renancaraujo) [reply lib/game.dart]: You are a human linter

File: lib/game.dart (hunk 6 above)
Comment (renancaraujo) [reply lib/game.dart]: image

File: lib/game.dart (hunk 1 above)
Comment (erickzanardo) [reply lib/game.dart]: If you want, I can help you test on Tales @renancaraujo

File: lib/game.dart (hunk 5 above)
Comment (erickzanardo) [reply lib/game.dart]: Is there a linter for Dart?

Comment (renancaraujo) [reply lib/game.dart]: `flutter analyze .`

Comment (luanpotter) [reply lib/game.dart]: `luanpotter --dart .`

File: lib/game.dart (hunk 2 above)
Comment (luanpotter) [reply lib/game.dart]: <space>{ (in case it was hard to see the space)

**Comment by renancaraujo** [comment]
@luanpotter a few responses

File: lib/game.dart (hunk 4 above)
Comment (luanpotter) [reply lib/game.dart]: I was talking about the comma, I misspelled the var name hehe

File: lib/game.dart (hunk 1 above)
Comment (renancaraujo) [reply lib/game.dart]: @erickzanardo would you do it?

Comment (erickzanardo) [reply lib/game.dart]: Yeah, could you rebase this branch with master? So I can test everything?

Comment (renancaraujo) [reply lib/game.dart]: @erickzanardo done

**Comment by erickzanardo** [comment]
@renancaraujo @luanpotter something is not right :/

Captura de Tela 2019-03-22 às 09 53 07

**Comment by renancaraujo** [comment]
@erickzanardo @luanpotter Made some changes and tested with the TRex. Please check it out.

**Comment by luanpotter** [comment]
Now we need to fix the lint hehehe :P

**Comment by erickzanardo** [comment]
Want me to test on tales?

Em dom, 24 de mar de 2019 07:25, Luan Nico <notifications@github.com>
escreveu:

**Comment by luanpotter** [comment]
Yes, please!

**Comment by luanpotter** [comment]
@renancaraujo we can fix conflicts and merge, please!

**Comment by renancaraujo** [comment]
@renancaraujo Yep, doing it asap

**Comment by luanpotter** [comment]
@renancaraujo way ahead of you ;)

**Comment by erickzanardo** [comment]
@renancaraujo @luanpotter just needs to update the CHANGELOG.md

**Comment by luanpotter** [comment]
Merged!

**Pull request merged**